rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp

```diff
@@ -298,6 +298,11 @@ __set_parameters_atomically_common(
   const OnParametersSetCallbackType & callback,
   bool allow_undeclared = false)
 {
+  // Set the parameter name before checks, for more desciprtive error messages
+  for (size_t i = 0; i < parameters.size(); ++i) {
+    const std::string & name = parameters[i].get_name();
+    parameter_infos[name].descriptor.name = parameters[i].get_name();
```

Comment:
see https://github.com/ros2/rclcpp/issues/1906#issuecomment-1081089117. i am not sure which tests are actually failing though.

Reply:
Thanks for context about the ParameterInfo structure - do you have a guess why this would cause tests to fail though?

Reply:
Dropping this approach and favoring 746d93b instead.

Reply:
> Thanks for context about the ParameterInfo structure - do you have a guess why this would cause tests to fail though?

no, not really. i have never tried this patch with my local environment yet.